### PR TITLE
Cleanup Condition Tests

### PR DESCRIPTION
**Pull request by KPrasch**
**Type of PR:**
Bugfix / Test Enhancement

**What**
- Ensure unit tests to not compile solidity
- Restores unit test suite execution speed to 500ms
- Adds a few more lingo serialization tests
- De-duplicates fixture definitions for conditions testing
- Converts loose test module constants to fixtures
- Automatic replacement of `"chain"` for test conditions
- Uses lowercase / snake-case conventions in fixtures

**Required reviews:** 
1

**Notes**
Co-Authored by @theref

### CODE REVIEW

File: tests/acceptance/network/test_middleware_retries.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import maya
+from web3.types import RPCEndpoint
+
+from nucypher.blockchain.middleware.retry import RetryRequestMiddleware
+from tests.constants import RPC_TOO_MANY_REQUESTS
+
+
+# TODO - since this test does exponential backoff it takes >= 2^1 = 2s, should we only run on CI?
+def test_request_with_retry_exponential_backoff():
```
Comment (derekpierre) [inline tests/acceptance/network/test_middleware_retries.py:11]: 👍 - nice to move this slower test to acceptance.

**Review by derekpierre — COMMENTED** [review]
Needs a rebase.

File: tests/unit/conditions/test_condition_lingo.py [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
-    clingo = ConditionLingo.from_list(CONDITIONS)
+def test_condition_lingo_repr(lingo):
+    clingo = ConditionLingo.from_list(lingo)
     clingo_string = f"{clingo}"
     assert f"{clingo.__class__.__name__}" in clingo_string
     assert f"id={clingo.id}" in clingo_string
     assert f"size={len(bytes(clingo))}" in clingo_string
+
+
+def test_condition_lingo_bug(custom_abi_with_multiple_parameters, mocker):
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply tests/unit/conditions/test_condition_lingo.py]: 👍🏻 done

File: tests/acceptance/network/test_middleware_retries.py (hunk 1 above)
Comment (KPrasch) [reply tests/acceptance/network/test_middleware_retries.py:11]: It's a bit out-of-scope in this PR, but we snuck it in, considering it was responsible for ~80% of the unit test runtime.

**Review by derekpierre — APPROVED** [review]
🎸

**Pull request merged**